packages/grafana-ui/src/components/Table/TableNG/TableNG.test.tsx

```diff
@@ -0,0 +1,1135 @@
+        // Look for a button inside the header
+        const sortButton = columnHeader.querySelector('button') || columnHeader;
+
+        // Click the sort button
+        fireEvent.click(sortButton);
+
+        // After clicking, the header should have an aria-sort attribute
+        const newSortAttribute = columnHeader.getAttribute('aria-sort');
+
+        // The sort attribute should have changed
+        expect(newSortAttribute).not.toBe(initialSortAttribute);
+
+        // The sort attribute should be either 'ascending' or 'descending'
+        expect(['ascending', 'descending']).toContain(newSortAttribute);
+
+        // Also verify the data is sorted by checking cell values
+        const cells = container.querySelectorAll('[role="gridcell"]');
+        const firstColumnCells = Array.from(cells).filter((_, index) => index % 2 === 0);
+
+        // Get the text content of the first column cells
+        const cellValues = firstColumnCells.map((cell) => cell.textContent);
+
+        // Verify we have values to check
+        expect(cellValues.length).toBeGreaterThan(0);
+
+        // Verify the values are in sorted order based on the aria-sort attribute
+        const sortedValues = [...cellValues].sort();
+
+        if (newSortAttribute === 'ascending') {
+          expect(JSON.stringify(cellValues)).toBe(JSON.stringify(sortedValues));
+        } else if (newSortAttribute === 'descending') {
+          expect(JSON.stringify(cellValues)).toBe(JSON.stringify([...sortedValues].reverse()));
+        }
+      }
+    });
+
+    it('cycles through ascending, descending, and no sort states', () => {
+      // Mock scrollIntoView
+      window.HTMLElement.prototype.scrollIntoView = jest.fn();
+
+      const { container } = render(
+        <TableNG enableVirtualization={false} data={createBasicDataFrame()} width={800} height={600} />
+      );
+
+      // Get the first column header
+      const columnHeader = container.querySelector('[role="columnheader"]');
+      expect(columnHeader).toBeInTheDocument();
+
+      if (columnHeader) {
+        const sortButton = columnHeader.querySelector('button') || columnHeader;
+
+        // Initial state - no sort
+        expect(columnHeader.getAttribute('aria-sort')).toBeNull();
+
+        // First click - should sort ascending
+        fireEvent.click(sortButton);
+        expect(columnHeader.getAttribute('aria-sort')).toBe('ascending');
+
+        // Second click - should sort descending
+        fireEvent.click(sortButton);
+        expect(columnHeader.getAttribute('aria-sort')).toBe('descending');
+
+        // Third click - should remove sort
+        fireEvent.click(sortButton);
+        expect(columnHeader.getAttribute('aria-sort')).toBeNull();
+      }
+    });
+
+    // TODO: The react-data-grid library does not have aria-sort support for multi-column
+    // sorting with shift key. We should open an issue to add this feature.
+    it.skip('supports multi-column sorting with shift key', () => {
```

Comment:
Can we remove the skip and instead look at the expected results, the actual values in the cells?